src/server/server/vineyard_server.cc

```diff
@@ -107,9 +107,13 @@ Status VineyardServer::Serve(StoreType const& bulk_store_type) {
     stream_store_ = nullptr;
   } else if (bulk_store_type_ == StoreType::kDefault) {
     bulk_store_ = std::make_shared<BulkStore>();
-    RETURN_ON_ERROR(bulk_store_->PreAllocate(
-        spec_["bulkstore_spec"]["memory_size"].get<size_t>()));
-
+    auto mem_limit = spec_["bulkstore_spec"]["memory_size"].get<size_t>();
+    RETURN_ON_ERROR(bulk_store_->PreAllocate(mem_limit));
+    // we should set the threshold as a portion of mem limit
+    // TODO: Maybe someday we should make SpillSize a configurable option
+    bulk_store_->SetMemSpillSize(mem_limit / 3);
```

Comment:
Make the spilling threshold a configurable value. For example, a spilling rate r (0<=r<1), and trigger spilling when used_memory < (mem_limit * r)